File: java/src/main/java/com/newnancity/newnanmanager/SessionContext.java
package com.newnancity.newnanmanager;

/** Server-instance session tuple used for request fencing. */
public final class SessionContext {
    private final String id;
    private final long epoch;

    public SessionContext(String id, long epoch) {
        if (id == null || id.trim().isEmpty()) {
            throw new IllegalArgumentException("session id must not be blank");
        }
        if (id.length() < 32 || id.length() > 64) {
            throw new IllegalArgumentException("session id length must be between 32 and 64 characters");
        }
        if (epoch <= 0L) {
            throw new IllegalArgumentException("session epoch must be greater than zero");
        }
        this.id = id;
        this.epoch = epoch;
    }

    public String getId() {
        return id;
    }

    public long getEpoch() {
        return epoch;
    }
}

File: kotlin/src/main/kotlin/com/nanmanager/bukkit/SessionContext.kt
package com.nanmanager.bukkit

import com.fasterxml.jackson.annotation.JsonProperty

/** Server session fencing headers issued when a server connection is opened. */
data class SessionContext(
    @JsonProperty("session_id")
    val id: String,
    @JsonProperty("session_epoch")
    val epoch: Long
) {
    init {
        require(id.isNotBlank()) { "session id must not be blank" }
        require(id.length in 32..64) { "session id length must be between 32 and 64 characters" }
        require(epoch > 0) { "session epoch must be greater than zero" }
    }
}
